test/index.js

```diff
@@ -563,69 +486,69 @@ describe('Mongo', () => {
 
             const mongo = new Mongo(options);
 
-            mongo.start((err) => {
-
+            try {
+                await mongo.start();
+            }
+            catch (err) {
                 expect(err).to.exist();
                 expect(err).to.be.instanceOf(Error);
-                done();
-            });
+            }
         });
 
-        it('connects with authentication', (done) => {
+        it('connects with authentication', async () => {
 
             const options = {
                 uri: 'mongodb://tester:secret@127.0.0.1:27017/?maxPoolSize=5',
                 partition: 'unit-testing'
             };
             const mongo = new Mongo(options);
 
-            mongo.start((err) => {
-
+            try {
+                await mongo.start();
+            }
+            catch (err) {
                 expect(err).to.not.exist();
-                done();
-            });
+            }
         });
 
-        it('sets isReady to true when the connection succeeds', (done) => {
+        it('connects with pending start', async () => {
 
             const options = {
-                uri: 'mongodb://127.0.0.1:27017/?maxPoolSize=5',
+                uri: 'mongodb://tester:secret@127.0.0.1:27017/?maxPoolSize=5',
                 partition: 'unit-testing'
             };
             const mongo = new Mongo(options);
+            mongo.startPending = true;
 
-            mongo.start((err) => {
-
+            try {
+                await mongo.start();
+            }
+            catch (err) {
                 expect(err).to.not.exist();
-                expect(mongo.isReady()).to.be.true();
-                done();
-            });
+            }
         });
 
-        it('calls any pending callbacks waiting for a start', (done) => {
```

Comment:
I think this should stay - as commented in start function its still important that start function resolves when actual connection starts even if called multiple times before connection is created.

Reply:
Yep, agree. With your explanation I see the need for it